### PR TITLE
URL values in raw tag editor override taginfo link with open in browser link

### PR DESCRIPTION
**Pull request by jackgrossman18**
ref #1769

### CODE REVIEW

File: modules/ui/tag_link_url.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            .data([0]);
+
+        _button = _button.enter()
+            .append('button')
+            .attr('class', 'tag-reference-button')
+            .attr('tabindex', -1)
+            .merge(_button);
+
+        _button
+            .attr('title', t('icons.open_link'))
```
Comment (brianhatchl) [inline modules/ui/tag_link_url.js:39]: We need to add the value for `icons.open_link` to core.yaml

File: modules/ui/raw_tag_editor.js [hunk 2]
```diff
@@ -175,10 +174,24 @@ export function uiRawTagEditor(context) {
...
-                row.call( select );
+                if (isNaN(d.value) && typeof d.value === 'string' && d.value.indexOf('http') === 0) {
+                    reference = uiTagLinkUrl({ url: d.value });
+
+                row.select('.inner-wrap')      // propagate bound data
+                .call(reference.button);
+
+                row.call(reference.body);
+
+                row.select('button.remove');   // propagate bound data
```
Comment (brianhatchl) [inline modules/ui/raw_tag_editor.js:200]: Not sure if we need this or can just call
`row.call(reference)`

the difference in syntax might be where this was plucked from iD `v2.14.`3 and hoot-ui is only on iD `v2.12.1`

File: modules/ui/raw_tag_editor.js [hunk 3]
```diff
@@ -339,6 +352,14 @@ export function uiRawTagEditor(context) {
                 _newRow = undefined;
             }
 
+            //Enable tag link url
+            if (this.value.indexOf('http') === 0) {
+                var reference = uiTagLinkUrl({ url: this.value });
+                var row = d3_select(this.parentNode.parentNode.parentNode);
+                row.select('.inner-wrap').call(reference.button);
+                row.call(reference.body);
+            }
```
Comment (brianhatchl) [inline modules/ui/raw_tag_editor.js]: When a new tag is added with a url value, this code is invoked, but it doesn't remove or update the existing tag copy button, leading to the ui appear like this:

Screenshot from 2020-05-19 17-33-57

When an existing feature with a url tag is selected (or another feature is selected and this feature then re-selected) the tag table renders fine.

**Review by brianhatchl — CHANGES_REQUESTED** [review]

File: modules/ui/raw_tag_editor.js (hunk 2 above)
Comment (jackgrossman18) [reply modules/ui/raw_tag_editor.js:200]: in testing removing body from line 188, it results in empty rows

remove_body

File: modules/ui/raw_tag_editor.js [hunk 4]
```diff
@@ -339,6 +363,14 @@ export function uiRawTagEditor(context) {
...
             }
 
+            // //Enable tag link url
+            // if (this.value.indexOf('http') === 0) {
+            //     var reference = uiTagLinkUrl({ url: this.value });
+            //     var row = d3_select(this.parentNode.parentNode.parentNode);
+            //     row.select('.inner-wrap').call(reference.button);
+            //     row.call(reference.body);
+            // }
+
```
Comment (brianhatchl) [inline modules/ui/raw_tag_editor.js]: remove if not needed

File: modules/ui/raw_tag_editor.js [hunk 5]
```diff
@@ -175,10 +175,34 @@ export function uiRawTagEditor(context) {
                     bindTypeahead(key, value);
                 }
 
+                var isRelation = (_entityID && context.entity(_entityID).type === 'relation');
+                var reference;
+
                 // Hoot: Override tag reference with tag copy
                 var select = uiTagSelectCopy(context);
 
                 row.call( select );
```
Comment (brianhatchl) [inline modules/ui/raw_tag_editor.js]: Should call this after other button types or move checkmark above column

image

**Comment by brianhatchl** [comment]
The border radius needs to be moved to the tag copy button
image

**Review by brianhatchl — APPROVED** [review]

**Pull request merged**